src/dnp3_python/dnp3station/outstation.py

```diff
@@ -550,11 +550,11 @@ class OutStationApplication:
 
     def __init__(
         self,
-        outstation_ip: str = "0.0.0.0",
-        port: int = 20000,
-        master_id: int = 2,
-        outstation_id: int = 1,
-        concurrency_hint: int = 1,
+        outstation_ip: str | None = "0.0.0.0",
```

Comment:
Explain why these are necessary as you are setting the default to a constant so they will never be none?

Reply:
This is to pass the type hint checking issue.

Say, 

When init the OutstationApplication(outstation_ip=outstation_ip), the outstation_ip is retrieved from terminal input, which can be None. And the type hint validation will complain about this.

Another option is use outstation_ip: str | None = None, then `if outstaion_ip is None: outstaion_ip = "0.0.0.0"`, which can be more tedious.